Developed Bubble sort.

rootDEV2990 opened the pull request:
Please help review my bubble sort in ruby. =)

T.Y.
-Miguel

majovanilla commented on sortby.rb:
```diff
@@ -0,0 +1,38 @@
+  array
+end
+
+def sorting(array)
+  sorted = true
+  (0...(array.count - 1)).each do |index|
+    if array[index] > array[index + 1]
+      sorted = false
+      array[index], array[index + 1] = [array[index + 1], array[index]]
+    end
+  end
+  [!sorted, array]
+end
```
In this line, you are returning true or false beside the sorted array. That shouldn't happen if you integrate both methods.

majovanilla replied:
This should be your main method and it should integrate the one on the top. Please make the necessary adjustments.

majovanilla commented on sortby.rb:
```diff
@@ -0,0 +1,38 @@
+def bubble_sort_items(array)
+  items_unsorted = true
+  items_unsorted, array = sorting(array) while items_unsorted
+  array
+end
```
What's this method for? You should only have two methods on this project: 
bubble_sort
bubble_sort_by
I recommend you to change the names to match those.
Also, delete the extra method and make sure you have only two that return the sorted array.

majovanilla commented on sortby.rb:
```diff
@@ -0,0 +1,38 @@
+  [!sorted, array]
+end
+
+def bubble_sort_by(array)
+  items_unsorted = true
```
This method doesn't take a block as an argument and order the elements using the block. Please make the necessary adjustments to fix it. Right now, when you pass a block, it's returning the array sorted but not with the order of the block passed.

majovanilla commented on README.md:
```diff
@@ -0,0 +1,57 @@
+Clone directory and run with Ruby. Open terminal inside project folder and type "ruby hello.rb" and hit enter.
+
+
+### Prerequisites
+
+Ruby installed on your computer to be able to read/run this file. 
+
+### Setup
+
+If your using Arch you can type in terminal.. 
+
+sudo pacman -S ruby --noconfirm 
+
+to install. if your OS is diffrent check out the link bellow.
+
+https://www.ruby-lang.org/en/downloads/
```
This part is not descriptive of your project. You don't have to use the prerequisites but you should give the user some guidance to run your methods.

rootDEV2990 commented:
Rewrote the bubble sort please help review i am still working on fixing rubocop errors

suretrust commented on sortby.rb:
```diff
@@ -0,0 +1,35 @@
+  loop do
+    swap = false
+    (loop_size - 1).times do |i|
+      if yield(array[i], array[i + 1]) > 0
+
+      array[i], array[i + 1] = array[i + 1], array[i]
+      swap = true
```
You are missing an `end` here.
```ruby
if (condition)
end
```

suretrust commented on sortby.rb:
```diff
@@ -0,0 +1,35 @@
+  array
+end
+
+input = [5, 3, 4, 9, 1, 8]
+p bubble_sort(input)
```
Let's remove this once you are done.

suretrust commented on sortby.rb:
```diff
@@ -0,0 +1,35 @@
+  array
+end
+
+inp = [1, 3, 4, 5, 8, 9]
+p bubble_sort_by(inp) { |first, second| second - first }
```
Let's remove this once you are done.

rootDEV2990 commented:
hi i fixed rubo cop and added missing ends to my if but i dont know what this means....
![Screen Shot 2020-04-29 at 2 55 40 PM](https://user-images.githubusercontent.com/60228072/80640875-a7b26500-8a29-11ea-8053-f765c4e6d767.png)

pelzolga123 commented:
## Hi Miguel! ![](https://user-images.githubusercontent.com/38801423/62908122-703d4480-bd44-11e9-9115-42025ab7f341.png) 
### Your project looks :+1: amazing!
Just one thing left:
* [ ] Make sure your `.rubocop.yml` file has the correct content. You can find original one [here](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml).

### Please kindly fix the above issue and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

rootDEV2990 commented:
Updated rubocop and fixed errors to request approval to merge to master.
